### PR TITLE
First 3 tasks from threads.

### PR DESCRIPTION
**Pull request by martinmilev**
Ready for review.

### CODE REVIEW

File: src/com/clouway/threads/task3/Demo.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  public void run() {
+    synchronized (this) {
+      for (int i = 1; i <= max; i++) {
+        try {
+          notifyAll();
+          wait();
+          Thread.sleep(500);
+        } catch (InterruptedException e) {}
+        System.out.println(Thread.currentThread().getName() + " " + i);
+        if(i==5){
```
Comment (mgenov) [inline src/com/clouway/threads/task3/Demo.java]: if(i==5) looks a little bit strange to me.

File: src/com/clouway/threads/task2/Demo.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+ * Running two threads simultaneously.
+ * First thread is counting forwards, second thread is counting backwards.
+ *
+ * @Author Martin Milev
+ * @Date 04.07.16.
+ */
+public class Demo {
+  public static void main(String[] args) {
+    int min = 10, max = 20;
+    Thread countUp = new Thread(new Counter(min, max, true));
```
Comment (mgenov) [inline src/com/clouway/threads/task2/Demo.java]: Looks like a different task: 

```
Да се направят 2 нишки броячи - броят от начална до крайна стойност.

На всяко отброяване да се извежда стойността на брояча.
На едната нишка се задава да брои до по-малка стойност от другата.
Когато една от нишките преброи до зададената стойност спира другата и завършва изпълнението си.
Да се пуснат нишките с малко различаващи се стойности и със стойности с голяма разлика.
```

File: src/com/clouway/threads/task3/Demo.java (hunk 1 above)
Comment (martinmilev) [inline src/com/clouway/threads/task3/Demo.java]: Otherwise the last thread waits for notification from no one and the program never finishes.

Comment (mgenov) [inline src/com/clouway/threads/task3/Demo.java]: Yeah, but there is better way for solving it.

File: src/com/clouway/threads/task2/Demo.java (hunk 2 above)
Comment (martinmilev) [inline src/com/clouway/threads/task2/Demo.java]: For example something like this?
```java
new Thread(new Counter(10, 50));
new Thread(new Counter(10, 35));
```

Comment (mgenov) [inline src/com/clouway/threads/task2/Demo.java]: Yeah, something similar but you have to consider how threads will communicate with each other.

Comment (mgenov) [inline src/com/clouway/threads/task2/Demo.java]: "Когато една от нишките преброи до зададената стойност спира другата и завършва изпълнението си."

File: src/com/clouway/threads/task3/Demo.java (hunk 1 above)
Comment (martinmilev) [inline src/com/clouway/threads/task3/Demo.java]: fixed?

File: src/com/clouway/threads/task3/Demo.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    Thread t2 = new Thread(counter);
+
+    t1.start();
+    t2.start();
+  }
+}
+
+class Counter implements Runnable {
+  private final int max;
+  private boolean b = true;
```
Comment (mgenov) [inline src/com/clouway/threads/task3/Demo.java]: b is not used as I can see.

File: src/com/clouway/threads/task2/Demo.java (hunk 2 above)
Comment (martinmilev) [inline src/com/clouway/threads/task2/Demo.java]: Done with the correct requirementс.

**Comment by mgenov** [comment]
LGTM !

**Pull request merged**